ci-operator/step-registry/mco/conf/day2/destroy-custom-mcp/mco-conf-day2-destroy-custom-mcp-commands.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+function set_proxy () {
+    if [ -s "${SHARED_DIR}/proxy-conf.sh" ]; then
+        echo "Setting the proxy ${SHARED_DIR}/proxy-conf.sh"
+        # shellcheck source=/dev/null
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "No proxy settings"
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running command: ${CMD}"
+    eval "${CMD}"
+}
+
+# waiting for mcp become a given status
+function wait_for_config_to_be_applied() {
+    local mcp_name="$1"
+    local expected_machine_count="$2"
+    local out MACHINECOUNT 
+
+    echo "Waiting for $mcp_name MachineConfigPool to finish updating..."
+    oc wait mcp "$mcp_name" --for='condition=UPDATED=True' --timeout=300s 2>/dev/null
+    
+    out="$(oc get machineconfigpools ${mcp_name} --no-headers)"
+    echo $out
+    MACHINECOUNT="$(echo "${out}" | awk '{print $6}')"
+
+    if [[ "${MACHINECOUNT}" != "${expected_machine_count}" ]]; then
+        run_command "oc get machineconfigpools"
+        run_command "oc get nodes"
+        echo "MCP ${mcp_name} is not ready for next step, exit..." && exit 1
+    fi
+}
+
+function remove_custom_mcp() {
+    local mcp_name="$1"
+    local mcp_node_label="node-role.kubernetes.io/${mcp_name}="
```

Comment:
cc: @sergiordlr @ptalgulk01

Reply:
`mcp_node_label` in `MCO_CONF_DAY2_CUSTOM_MCP` is used to selected which nodes would be label with new label, refer to https://github.com/openshift/release/blob/master/ci-operator/step-registry/mco/conf/day2/create-custom-mcp/mco-conf-day2-create-custom-mcp-commands.sh#L81, So I think it is doing right things.

Reply:
yes, the mcp label created by step `create-custom-mcp` works well with this step, but this step can be used independently. the function depends on the input param not step `create-custom-mcp`, if input mcp is created with other label. i.e. not `node-role.kubernetes.io/$name`. the mcp removal will be failed.

Reply:
The reason of naming the step as `destroy-custom-mcp` (the opposite of `create-custom-mcp`) is just because we want the two steps can be used as a pair, so that we can easily understand and maintain them. 

If some custom mcp is not crated by `create-custom-mcp`, but some other step, the corresponding destroy step should also be managed by other step, but not by `create-destroy-mcp`.

Reply:
this step is under dir mco, from ci-operator perspective, it can be used independently. there is no rule can block the user to use this step like you said.

Reply:
We can update [document](https://github.com/openshift/release/pull/55328/files#diff-0a77883fc30e6ee928ed1e8ed59ce465fa7ea29afbf512282337b3a90dc83549R15) to indicate the purpose of this step. @JianLi-RH WDYT?

Reply:
@jianlinliu @rioliu-rh 
Shall we add a new field in the json, for example 'custom_node_label':
        '[{
          "mcp_name": "infra", 
          "mcp_node_num": "1", 
          "mcp_node_label": "node-role.kubernetes.io/worker=,kubernetes.io/arch=arm64",
          "custom_node_label":"xxx"
        }]'. 
if custom_node_label does not exist or its value is empty, then we use `"node-role.kubernetes.io/${mcp_name}="` otherwise use its value to find and remove node lable.

Reply:
IMO theorically we only need mcp name, both `mcp_node_num` and `mcp_node_label` can be got from resource mcp

Reply:
More detail discussion can be found in https://redhat-internal.slack.com/archives/CF8SMALS1/p1725852004587709, the conclusion is:
1.  decouple create mcp and destory mcp 
2. add a new env in step `mco-conf-day2-destroy-custom-mcp`: MCO_CONF_DAY2_CUSTOM_MCP_TO_BE_DESTROYED
3. read node label from nodeSelector from mcp yaml